consensus/validation.go

```diff
@@ -395,6 +406,15 @@ func (vc *ValidationContext) validFileContractResolutions(txn types.Transaction)
 	return nil
 }
 
+func (vc *ValidationContext) validAttestations(txn types.Transaction) error {
+	for i, a := range txn.Attestations {
+		if !a.PublicKey.VerifyHash(vc.AttestationSigHash(a), a.Signature) {
```

Comment:
Should we prevent empty keys and values on attestations as well?

Reply:
hmm, probably. I can imagine a use case for empty `Data`, but not for empty `Key`.